### ISSUE TITLE
Connections Dropped Under macOS 10.12 Sierra

### ISSUE THREAD

**Issue by ghalfacree** [syncthing/syncthing#3396]
Syncthing v0.13.10 appears to dislike the macOS 10.12 Sierra beta release. The application runs, connects, appears to be working, then disconnects after a handful of minutes. No reconnection is attempted, and no errors beyond the disconnections themselves appear in the logs. Other devices in the pool remain connected to each other. Restarting Syncthing on the macOS Sierra laptop reconnects, but again fails after a handful of minutes.

Reproduction:
Upgrade a Mac to macOS 10.12 Sierra Beta 2 (available publicly via the Apple Software Beta Program.) Run Syncthing v0.13.10.

Expected outcome:
Syncthing connects to the shares, synchronises files, everything is hunky-dory.

Actual outcome:
Syncthing connects to the shares, then disconnects from all shares in under ten minutes. No reconnection is possible without restarting the application.

Log text:
[3WAOH] 17:02:29 INFO: Device XXXXXXX-XXXXXXX-XXXXXXX-XXXXXXX-XXXXXXX-XXXXXXX-XXXXXXX-XXXXXXX is "bender.halfacree.co.uk" at [dynamic]
[3WAOH] 17:02:29 INFO: GUI and API listening on 127.0.0.1:8080
[3WAOH] 17:02:29 INFO: Access the GUI via the following URL: http://127.0.0.1:8080/
[3WAOH] 17:02:29 INFO: Starting usage reporting
[3WAOH] 17:02:32 INFO: Established secure connection to YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY at 192.168.0.79:22000-192.168.0.50:36634 (TCP (Server))
[3WAOH] 17:02:32 INFO: Device YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY client is "syncthing v0.13.10" named "trioptimum"
[3WAOH] 17:02:32 INFO: Completed initial scan (rw) of folder Dropbox
[3WAOH] 17:02:32 INFO: Connected to already connected device (YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY)
[3WAOH] 17:02:32 INFO: Established secure connection to ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ at 192.168.0.79:51974-192.168.0.8:22000 (TCP (Client))
[3WAOH] 17:02:32 INFO: Device ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ client is "syncthing v0.13.10" named "Altair"
[3WAOH] 17:10:04 INFO: Connection to ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ-ZZZZZZZ closed: EOF
[3WAOH] 17:10:06 INFO: Connection to YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY-YYYYYYY closed: EOF

### Version Information

Syncthing Version: v0.13.10
OS Version: macOS 10.12 Sierra Beta 2

**Comment by AudriusButkevicius** [comment]
I suspect it's some new magic app nap thing they invented.

**Comment by calmh** [comment]
Yeah, I'd reword this as Sierra not likin Syncthing since one has changed and not the other. I would expect this to be resolved before it's released, but I'll test it when I upgrade some time later in the beta cycle.

**Comment by calmh** [comment]
Resolved by compiling with Go 1.6.3
